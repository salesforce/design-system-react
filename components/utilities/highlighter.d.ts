declare module '@salesforce/design-system-react/components/utilities/highlighter' {
	import React from 'react';
	type Props = {
		/**
		 * The full string to display.
		 */
		children?: string | number | boolean | React.ReactNode;
		className?: string;
		/**
		 * The string of text (or Regular Expression) to highlight.
		 */
		search?: PropTypes.any;
	};

	function Component(props: Props): JSX.Element;
	export default Component;
}
