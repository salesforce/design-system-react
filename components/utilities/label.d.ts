declare module '@salesforce/design-system-react/components/utilities/label' {
	import React from 'react';
	type Props = {
		/*
		 * Assistive Text to use instead of a visible label
		 */
		assistiveText?: Record<string, any>;
		/**
		 * Class names to be added to the label
		 */
		className?: any[] | Record<string, any> | string;
		/*
		 * Id of the input associated with this label
		 */
		htmlFor?: string;
		/*
		 * Input Label
		 */
		label?: string;
		/*
		 * Applies label styling for a required form element
		 */
		required?: boolean;
		/**
		 * Changes markup of label.
		 */
		variant?: 'base' | 'static';
	};

	function Component(props: Props): JSX.Element;
	export default Component;
}
