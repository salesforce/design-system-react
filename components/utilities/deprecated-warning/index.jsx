/* Copyright (c) 2015-present, salesforce.com, inc. All rights reserved */
/* Licensed under BSD 3-Clause - see LICENSE.txt or git.io/sfdc-license */

// ### React
import React from 'react';

// ## Constants
import { DEPRECATED_WARNING } from '../../../utilities/constants';

import Alert from '../../alert';
import AlertContainer from '../../alert/container';

/**
 * A utility component that is used to highlight a deprecated component.
 */
const DeprecatedWarning = () => (
	<div className="slds-p-top_x-large">
		<AlertContainer>
			<Alert
				labels={{
					heading: (
						<span>
							<strong>Deprecated component.</strong> No fit and finish bug fixes
							will be accepted for this component.
						</span>
					),
				}}
				variant="error"
			/>
		</AlertContainer>
	</div>
);

// ### Display Name
DeprecatedWarning.displayName = DEPRECATED_WARNING;

export default DeprecatedWarning;
