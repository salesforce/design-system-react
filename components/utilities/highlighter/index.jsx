/* Copyright (c) 2015-present, salesforce.com, inc. All rights reserved */
/* Licensed under BSD 3-Clause - see LICENSE.txt or git.io/sfdc-license */

// ### React
import React from 'react';
import PropTypes from 'prop-types';

// ### ReactHighlighter
import ReactHighlighter from 'react-highlighter';

// ## Constants
import { HIGHLIGHTER } from '../../../utilities/constants';

/**
 * A utility component that highlights occurrences of a particular pattern in its contents.
 */
const Highlighter = (props) => {
	if (props.search) {
		let children;
		if (typeof props.children === 'string') {
			children = (
				<ReactHighlighter
					className={props.className}
					matchClass={null}
					matchElement="mark"
					search={props.search}
					title={props.children}
				>
					{props.children}
				</ReactHighlighter>
			);
		} else {
			const findString = (nodeArr) =>
				nodeArr.map((element) => {
					let newElement;
					if (typeof element === 'string') {
						newElement = (
							<ReactHighlighter
								key={element}
								className={props.className}
								matchClass={null}
								matchElement="mark"
								search={props.search}
								title={element}
							>
								{element}
							</ReactHighlighter>
						);
					} else {
						newElement = element;
					}
					return newElement;
				});

			if (props.children.props) {
				const node = props.children.props.children;
				children = node instanceof Array ? findString(node) : node;
			}
		}

		return <span>{children}</span>;
	}

	if (typeof props.children === 'string') {
		return (
			<span className={props.className} title={props.children}>
				{props.children}
			</span>
		);
	}

	return <span className={props.className}>{props.children}</span>;
};

// ### Display Name
Highlighter.displayName = HIGHLIGHTER;

// ### Prop Types
Highlighter.propTypes = {
	/**
	 * The full string to display.
	 */
	children: PropTypes.oneOfType([
		PropTypes.string,
		PropTypes.number,
		PropTypes.bool,
		PropTypes.node,
	]),
	className: PropTypes.string,
	/**
	 * The string of text (or Regular Expression) to highlight.
	 */
	search: PropTypes.any,
};

export default Highlighter;
