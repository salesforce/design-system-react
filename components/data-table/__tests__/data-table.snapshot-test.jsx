/* eslint-env jest */
import {
	testDOMandHTML,
	testImageSnapshot
} from '../../../tests/snapshot-helpers';

import Basic from '../__examples__/basic';
import Advanced from '../__examples__/advanced';

import { DATA_TABLE } from '../../../utilities/constants';

describe(DATA_TABLE, () => {
	test('DOM & HTML Snapshots', () => {
		testDOMandHTML(Basic);
		testDOMandHTML(Advanced);
	});

	test('Basic Image Snapshot', async () => {
		await testImageSnapshot(DATA_TABLE, 'Basic');
	});

	test('Advanced Image Snapshot', async () => {
		await testImageSnapshot(DATA_TABLE, 'Advanced');
	});
});
