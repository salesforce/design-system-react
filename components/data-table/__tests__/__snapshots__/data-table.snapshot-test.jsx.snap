// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DataTable Advanced HTML Snapshot 1`] = `
"<div>
  <table class=\\"slds-table slds-table--fixed-layout slds-table--bordered\\" id=\\"DataTableExample-2\\" role=\\"grid\\">
    <thead>
      <tr class=\\"slds-text-title--caps\\">
        <th class=\\"slds-text-align--right\\" scope=\\"col\\" style=\\"width:3.25rem;\\">
          <div class=\\"slds-th__action slds-th__action--form\\">
            <div class=\\"slds-form-element\\">
              <div class=\\"slds-form-element__control\\"><span class=\\"slds-checkbox\\"><input type=\\"checkbox\\" id=\\"DataTableExample-2-SLDSDataTableHead-SelectAll\\" name=\\"SelectAll\\"/><label class=\\"slds-checkbox__label\\" for=\\"DataTableExample-2-SLDSDataTableHead-SelectAll\\"><span class=\\"slds-checkbox--faux\\"></span>
                <span
                  class=\\"slds-assistive-text\\">Select all rows</span>
                  </label>
                  </span>
              </div>
            </div>
          </div>
        </th>
        <th aria-sort=\\"ascending\\" class=\\"slds-is-sortable slds-is-sorted\\" focusable=\\"true\\" scope=\\"col\\" style=\\"width:10rem;\\"><a href=\\"javascript:void(0)\\" class=\\"slds-th__action slds-text-link--reset\\" tabindex=\\"0\\"><span class=\\"slds-assistive-text\\">Sort </span><span class=\\"slds-truncate\\" title=\\"Name\\">Name</span><span><svg aria-hidden=\\"true\\" class=\\"slds-is-sortable__icon slds-icon slds-icon--x-small slds-icon-text-default\\"><use xlink:href=\\"/assets/icons/utility-sprite/svg/symbols.svg#arrowup\\"></use></svg></span><span class=\\"slds-assistive-text\\" aria-live=\\"assertive\\" aria-atomic=\\"true\\">Sorted Ascending</span></a></th>
        <th
          class=\\"\\" focusable=\\"false\\" scope=\\"col\\" style=\\"width:8rem;\\">
          <div class=\\"slds-truncate\\">Account Name</div>
          </th>
          <th class=\\"\\" focusable=\\"false\\" scope=\\"col\\">
            <div class=\\"slds-truncate\\">Close Date</div>
          </th>
          <th class=\\"\\" focusable=\\"false\\" scope=\\"col\\">
            <div class=\\"slds-truncate\\">Stage</div>
          </th>
          <th class=\\"slds-is-sortable\\" focusable=\\"true\\" scope=\\"col\\"><a href=\\"javascript:void(0)\\" class=\\"slds-th__action slds-text-link--reset\\" tabindex=\\"0\\"><span class=\\"slds-assistive-text\\">Sort </span><span class=\\"slds-truncate\\" title=\\"Confidence\\">Confidence</span><span><svg aria-hidden=\\"true\\" class=\\"slds-is-sortable__icon slds-icon slds-icon--x-small slds-icon-text-default\\"><use xlink:href=\\"/assets/icons/utility-sprite/svg/symbols.svg#arrowup\\"></use></svg></span></a></th>
          <th
            class=\\"\\" focusable=\\"false\\" scope=\\"col\\">
            <div class=\\"slds-truncate\\">Amount</div>
            </th>
            <th class=\\"\\" focusable=\\"false\\" scope=\\"col\\">
              <div class=\\"slds-truncate\\">Contact</div>
            </th>
            <th scope=\\"col\\" style=\\"width:3.25rem;\\">
              <div class=\\"slds-th__action\\"><span class=\\"slds-assistive-text\\">Actions</span></div>
            </th>
      </tr>
    </thead>
    <tbody>
      <tr class=\\"slds-hint-parent\\">
        <td role=\\"gridcell\\" class=\\"slds-text-align--right\\" data-label=\\"Select Row\\" style=\\"width:3.25rem;\\">
          <div class=\\"slds-form-element\\">
            <div class=\\"slds-form-element__control\\"><span class=\\"slds-checkbox\\"><input type=\\"checkbox\\" id=\\"DataTableExample-2-SLDSDataTableRow-8IKZHZZV80-SelectRow\\" name=\\"SelectRow\\"/><label class=\\"slds-checkbox__label\\" for=\\"DataTableExample-2-SLDSDataTableRow-8IKZHZZV80-SelectRow\\"><span class=\\"slds-checkbox--faux\\"></span>
              <span
                class=\\"slds-assistive-text\\">Select row</span>
                </label>
                </span>
            </div>
          </div>
        </td>
        <th role=\\"gridcell\\" style=\\"width:10rem;\\">
          <div class=\\"slds-truncate\\" title=\\"Acme - 1,200 Widgets\\"><a href=\\"javascript:void(0);\\">Acme - 1,200 Widgets</a></div>
        </th>
        <td role=\\"gridcell\\" style=\\"width:8rem;\\">
          <div class=\\"slds-truncate\\" title=\\"Acme\\">Acme</div>
        </td>
        <td role=\\"gridcell\\">
          <div class=\\"slds-truncate\\" title=\\"4/10/15\\">4/10/15</div>
        </td>
        <td role=\\"gridcell\\">
          <div class=\\"slds-truncate\\" title=\\"Value Proposition\\">Value Proposition</div>
        </td>
        <td role=\\"gridcell\\">
          <div class=\\"slds-truncate\\" title=\\"70%\\">70%</div>
        </td>
        <td role=\\"gridcell\\">
          <div class=\\"slds-truncate\\" title=\\"$25,000,000\\">$25,000,000</div>
        </td>
        <td role=\\"gridcell\\">
          <div class=\\"slds-truncate\\" title=\\"jrogers@acme.com\\"><a href=\\"javascript:void(0);\\">jrogers@acme.com</a></div>
        </td>
        <td class=\\"\\" data-label=\\"Actions\\" style=\\"width:3.25rem;\\">
          <div class=\\"slds-dropdown-trigger slds-dropdown-trigger--click\\" id=\\"DataTableExample-2-SLDSDataTableRow-8IKZHZZV80-SLDSDataTableRowActions\\"><button aria-expanded=\\"false\\" aria-haspopup=\\"true\\" class=\\"slds-button slds-button--icon-border-filled slds-button--icon-small ignore-click-DataTableExample-2-SLDSDataTableRow-8IKZHZZV80-SLDSDataTableRowActions slds-button--icon-x-small\\" tabindex=\\"0\\"
              type=\\"button\\"><svg aria-hidden=\\"true\\" class=\\"slds-button__icon slds-button__icon--hint\\"><use xlink:href=\\"/assets/icons/utility-sprite/svg/symbols.svg#down\\"></use></svg><span class=\\"slds-assistive-text\\">Actions</span></button></div>
        </td>
      </tr>
      <tr class=\\"slds-hint-parent\\">
        <td role=\\"gridcell\\" class=\\"slds-text-align--right\\" data-label=\\"Select Row\\" style=\\"width:3.25rem;\\">
          <div class=\\"slds-form-element\\">
            <div class=\\"slds-form-element__control\\"><span class=\\"slds-checkbox\\"><input type=\\"checkbox\\" id=\\"DataTableExample-2-SLDSDataTableRow-5GJOOOPWU7-SelectRow\\" name=\\"SelectRow\\"/><label class=\\"slds-checkbox__label\\" for=\\"DataTableExample-2-SLDSDataTableRow-5GJOOOPWU7-SelectRow\\"><span class=\\"slds-checkbox--faux\\"></span>
              <span
                class=\\"slds-assistive-text\\">Select row</span>
                </label>
                </span>
            </div>
          </div>
        </td>
        <th role=\\"gridcell\\" style=\\"width:10rem;\\">
          <div class=\\"slds-truncate\\" title=\\"Acme - 200 Widgets\\"><a href=\\"javascript:void(0);\\">Acme - 200 Widgets</a></div>
        </th>
        <td role=\\"gridcell\\" style=\\"width:8rem;\\">
          <div class=\\"slds-truncate\\" title=\\"Acme\\">Acme</div>
        </td>
        <td role=\\"gridcell\\">
          <div class=\\"slds-truncate\\" title=\\"1/31/15\\">1/31/15</div>
        </td>
        <td role=\\"gridcell\\">
          <div class=\\"slds-truncate\\" title=\\"Prospecting\\">Prospecting</div>
        </td>
        <td role=\\"gridcell\\">
          <div class=\\"slds-truncate\\" title=\\"30%\\">30%</div>
        </td>
        <td role=\\"gridcell\\">
          <div class=\\"slds-truncate\\" title=\\"$5,000,000\\">$5,000,000</div>
        </td>
        <td role=\\"gridcell\\">
          <div class=\\"slds-truncate\\" title=\\"bob@acme.com\\"><a href=\\"javascript:void(0);\\">bob@acme.com</a></div>
        </td>
        <td class=\\"\\" data-label=\\"Actions\\" style=\\"width:3.25rem;\\">
          <div class=\\"slds-dropdown-trigger slds-dropdown-trigger--click\\" id=\\"DataTableExample-2-SLDSDataTableRow-5GJOOOPWU7-SLDSDataTableRowActions\\"><button aria-expanded=\\"false\\" aria-haspopup=\\"true\\" class=\\"slds-button slds-button--icon-border-filled slds-button--icon-small ignore-click-DataTableExample-2-SLDSDataTableRow-5GJOOOPWU7-SLDSDataTableRowActions slds-button--icon-x-small\\" tabindex=\\"0\\"
              type=\\"button\\"><svg aria-hidden=\\"true\\" class=\\"slds-button__icon slds-button__icon--hint\\"><use xlink:href=\\"/assets/icons/utility-sprite/svg/symbols.svg#down\\"></use></svg><span class=\\"slds-assistive-text\\">Actions</span></button></div>
        </td>
      </tr>
      <tr class=\\"slds-hint-parent slds-is-selected\\">
        <td role=\\"gridcell\\" class=\\"slds-text-align--right\\" data-label=\\"Select Row\\" style=\\"width:3.25rem;\\">
          <div class=\\"slds-form-element\\">
            <div class=\\"slds-form-element__control\\"><span class=\\"slds-checkbox\\"><input type=\\"checkbox\\" checked=\\"\\" id=\\"DataTableExample-2-SLDSDataTableRow-8IKZHZZV81-SelectRow\\" name=\\"SelectRow\\"/><label class=\\"slds-checkbox__label\\" for=\\"DataTableExample-2-SLDSDataTableRow-8IKZHZZV81-SelectRow\\"><span class=\\"slds-checkbox--faux\\"></span>
              <span
                class=\\"slds-assistive-text\\">Select row</span>
                </label>
                </span>
            </div>
          </div>
        </td>
        <th role=\\"gridcell\\" style=\\"width:10rem;\\">
          <div class=\\"slds-truncate\\" title=\\"salesforce.com - 1,000 Widgets\\"><a href=\\"javascript:void(0);\\">salesforce.com - 1,000 Widgets</a></div>
        </th>
        <td role=\\"gridcell\\" style=\\"width:8rem;\\">
          <div class=\\"slds-truncate\\" title=\\"salesforce.com\\">salesforce.com</div>
        </td>
        <td role=\\"gridcell\\">
          <div class=\\"slds-truncate\\" title=\\"1/31/15 3:45PM\\">1/31/15 3:45PM</div>
        </td>
        <td role=\\"gridcell\\">
          <div class=\\"slds-truncate\\" title=\\"Id. Decision Makers\\">Id. Decision Makers</div>
        </td>
        <td role=\\"gridcell\\">
          <div class=\\"slds-truncate\\" title=\\"60%\\">60%</div>
        </td>
        <td role=\\"gridcell\\">
          <div class=\\"slds-truncate\\" title=\\"$25,000\\">$25,000</div>
        </td>
        <td role=\\"gridcell\\">
          <div class=\\"slds-truncate\\" title=\\"nathan@salesforce.com\\"><a href=\\"javascript:void(0);\\">nathan@salesforce.com</a></div>
        </td>
        <td class=\\"\\" data-label=\\"Actions\\" style=\\"width:3.25rem;\\">
          <div class=\\"slds-dropdown-trigger slds-dropdown-trigger--click\\" id=\\"DataTableExample-2-SLDSDataTableRow-8IKZHZZV81-SLDSDataTableRowActions\\"><button aria-expanded=\\"false\\" aria-haspopup=\\"true\\" class=\\"slds-button slds-button--icon-border-filled slds-button--icon-small ignore-click-DataTableExample-2-SLDSDataTableRow-8IKZHZZV81-SLDSDataTableRowActions slds-button--icon-x-small\\" tabindex=\\"0\\"
              type=\\"button\\"><svg aria-hidden=\\"true\\" class=\\"slds-button__icon slds-button__icon--hint\\"><use xlink:href=\\"/assets/icons/utility-sprite/svg/symbols.svg#down\\"></use></svg><span class=\\"slds-assistive-text\\">Actions</span></button></div>
        </td>
      </tr>
    </tbody>
  </table>
</div>"
`;

exports[`DataTable Basic HTML Snapshot 1`] = `
"<div style=\\"overflow:auto;\\">
  <h3 class=\\"slds-text-heading--medium slds-m-vertical--medium\\">Default Fluid Layout</h3>
  <table class=\\"slds-table slds-table--bordered slds-table--cell-buffer\\" id=\\"DataTableExample-1-default\\">
    <thead>
      <tr class=\\"slds-text-title--caps\\">
        <th class=\\"\\" focusable=\\"false\\" scope=\\"col\\">
          <div class=\\"slds-truncate\\">Opportunity Name</div>
        </th>
        <th class=\\"\\" focusable=\\"false\\" scope=\\"col\\">
          <div class=\\"slds-truncate\\">Account Name</div>
        </th>
        <th class=\\"\\" focusable=\\"false\\" scope=\\"col\\">
          <div class=\\"slds-truncate\\">Close Date</div>
        </th>
        <th class=\\"\\" focusable=\\"false\\" scope=\\"col\\">
          <div class=\\"slds-truncate\\">Stage</div>
        </th>
        <th class=\\"\\" focusable=\\"false\\" scope=\\"col\\">
          <div class=\\"slds-truncate\\">Confidence</div>
        </th>
        <th class=\\"\\" focusable=\\"false\\" scope=\\"col\\">
          <div class=\\"slds-truncate\\">Amount</div>
        </th>
        <th class=\\"\\" focusable=\\"false\\" scope=\\"col\\">
          <div class=\\"slds-truncate\\">Contact</div>
        </th>
      </tr>
    </thead>
    <tbody>
      <tr class=\\"\\">
        <td>
          <div class=\\"\\" title=\\"Cloudhub\\"><a href=\\"javascript:void(0);\\">Cloudhub</a></div>
        </td>
        <td>
          <div class=\\"\\" title=\\"Cloudhub\\">Cloudhub</div>
        </td>
        <td>
          <div class=\\"\\" title=\\"4/14/2015\\">4/14/2015</div>
        </td>
        <td>
          <div class=\\"\\" title=\\"Prospecting\\">Prospecting</div>
        </td>
        <td>
          <div class=\\"\\" title=\\"20%\\">20%</div>
        </td>
        <td>
          <div class=\\"\\" title=\\"$25k\\">$25k</div>
        </td>
        <td>
          <div class=\\"\\" title=\\"jrogers@cloudhub.com\\"><a href=\\"javascript:void(0);\\">jrogers@cloudhub.com</a></div>
        </td>
      </tr>
      <tr class=\\"\\">
        <td>
          <div class=\\"\\" title=\\"Cloudhub + Anypoint Connectors\\"><a href=\\"javascript:void(0);\\">Cloudhub + Anypoint Connectors</a></div>
        </td>
        <td>
          <div class=\\"\\" title=\\"Cloudhub\\">Cloudhub</div>
        </td>
        <td>
          <div class=\\"\\" title=\\"4/14/2015\\">4/14/2015</div>
        </td>
        <td>
          <div class=\\"\\" title=\\"Prospecting\\">Prospecting</div>
        </td>
        <td>
          <div class=\\"\\" title=\\"20%\\">20%</div>
        </td>
        <td>
          <div class=\\"\\" title=\\"$25k\\">$25k</div>
        </td>
        <td>
          <div class=\\"\\" title=\\"jrogers@cloudhub.com\\"><a href=\\"javascript:void(0);\\">jrogers@cloudhub.com</a></div>
        </td>
      </tr>
      <tr class=\\"\\">
        <td>
          <div class=\\"\\" title=\\"Cloudhub\\"><a href=\\"javascript:void(0);\\">Cloudhub</a></div>
        </td>
        <td>
          <div class=\\"\\" title=\\"Cloudhub\\">Cloudhub</div>
        </td>
        <td>
          <div class=\\"\\" title=\\"4/14/2015\\">4/14/2015</div>
        </td>
        <td>
          <div class=\\"\\" title=\\"Prospecting\\">Prospecting</div>
        </td>
        <td>
          <div class=\\"\\" title=\\"20%\\">20%</div>
        </td>
        <td>
          <div class=\\"\\" title=\\"$25k\\">$25k</div>
        </td>
        <td>
          <div class=\\"\\" title=\\"jrogers@cloudhub.com\\"><a href=\\"javascript:void(0);\\">jrogers@cloudhub.com</a></div>
        </td>
      </tr>
    </tbody>
  </table>
  <h3 class=\\"slds-text-heading--medium slds-m-vertical--medium\\">Striped</h3>
  <table class=\\"slds-table slds-table--bordered slds-table--cell-buffer slds-table--striped\\" id=\\"DataTableExample-1-striped\\">
    <thead>
      <tr class=\\"slds-text-title--caps\\">
        <th class=\\"\\" focusable=\\"false\\" scope=\\"col\\">
          <div class=\\"slds-truncate\\">Opportunity Name</div>
        </th>
        <th class=\\"\\" focusable=\\"false\\" scope=\\"col\\">
          <div class=\\"slds-truncate\\">Account Name</div>
        </th>
        <th class=\\"\\" focusable=\\"false\\" scope=\\"col\\">
          <div class=\\"slds-truncate\\">Close Date</div>
        </th>
        <th class=\\"\\" focusable=\\"false\\" scope=\\"col\\">
          <div class=\\"slds-truncate\\">Stage</div>
        </th>
        <th class=\\"\\" focusable=\\"false\\" scope=\\"col\\">
          <div class=\\"slds-truncate\\">Confidence</div>
        </th>
        <th class=\\"\\" focusable=\\"false\\" scope=\\"col\\">
          <div class=\\"slds-truncate\\">Amount</div>
        </th>
        <th class=\\"\\" focusable=\\"false\\" scope=\\"col\\">
          <div class=\\"slds-truncate\\">Contact</div>
        </th>
      </tr>
    </thead>
    <tbody>
      <tr class=\\"\\">
        <td>
          <div class=\\"\\" title=\\"Cloudhub\\"><a href=\\"javascript:void(0);\\">Cloudhub</a></div>
        </td>
        <td>
          <div class=\\"\\" title=\\"Cloudhub\\">Cloudhub</div>
        </td>
        <td>
          <div class=\\"\\" title=\\"4/14/2015\\">4/14/2015</div>
        </td>
        <td>
          <div class=\\"\\" title=\\"Prospecting\\">Prospecting</div>
        </td>
        <td>
          <div class=\\"\\" title=\\"20%\\">20%</div>
        </td>
        <td>
          <div class=\\"\\" title=\\"$25k\\">$25k</div>
        </td>
        <td>
          <div class=\\"\\" title=\\"jrogers@cloudhub.com\\"><a href=\\"javascript:void(0);\\">jrogers@cloudhub.com</a></div>
        </td>
      </tr>
      <tr class=\\"\\">
        <td>
          <div class=\\"\\" title=\\"Cloudhub + Anypoint Connectors\\"><a href=\\"javascript:void(0);\\">Cloudhub + Anypoint Connectors</a></div>
        </td>
        <td>
          <div class=\\"\\" title=\\"Cloudhub\\">Cloudhub</div>
        </td>
        <td>
          <div class=\\"\\" title=\\"4/14/2015\\">4/14/2015</div>
        </td>
        <td>
          <div class=\\"\\" title=\\"Prospecting\\">Prospecting</div>
        </td>
        <td>
          <div class=\\"\\" title=\\"20%\\">20%</div>
        </td>
        <td>
          <div class=\\"\\" title=\\"$25k\\">$25k</div>
        </td>
        <td>
          <div class=\\"\\" title=\\"jrogers@cloudhub.com\\"><a href=\\"javascript:void(0);\\">jrogers@cloudhub.com</a></div>
        </td>
      </tr>
      <tr class=\\"\\">
        <td>
          <div class=\\"\\" title=\\"Cloudhub\\"><a href=\\"javascript:void(0);\\">Cloudhub</a></div>
        </td>
        <td>
          <div class=\\"\\" title=\\"Cloudhub\\">Cloudhub</div>
        </td>
        <td>
          <div class=\\"\\" title=\\"4/14/2015\\">4/14/2015</div>
        </td>
        <td>
          <div class=\\"\\" title=\\"Prospecting\\">Prospecting</div>
        </td>
        <td>
          <div class=\\"\\" title=\\"20%\\">20%</div>
        </td>
        <td>
          <div class=\\"\\" title=\\"$25k\\">$25k</div>
        </td>
        <td>
          <div class=\\"\\" title=\\"jrogers@cloudhub.com\\"><a href=\\"javascript:void(0);\\">jrogers@cloudhub.com</a></div>
        </td>
      </tr>
    </tbody>
  </table>
  <h3 class=\\"slds-text-heading--medium slds-m-vertical--medium\\">No Row Hover</h3>
  <table class=\\"slds-table slds-table--bordered slds-table--cell-buffer slds-no-row-hover\\" id=\\"DataTableExample-noRowHover\\">
    <thead>
      <tr class=\\"slds-text-title--caps\\">
        <th class=\\"\\" focusable=\\"false\\" scope=\\"col\\">
          <div class=\\"slds-truncate\\">Opportunity Name</div>
        </th>
        <th class=\\"\\" focusable=\\"false\\" scope=\\"col\\">
          <div class=\\"slds-truncate\\">Account Name</div>
        </th>
        <th class=\\"\\" focusable=\\"false\\" scope=\\"col\\">
          <div class=\\"slds-truncate\\">Close Date</div>
        </th>
        <th class=\\"\\" focusable=\\"false\\" scope=\\"col\\">
          <div class=\\"slds-truncate\\">Stage</div>
        </th>
        <th class=\\"\\" focusable=\\"false\\" scope=\\"col\\">
          <div class=\\"slds-truncate\\">Confidence</div>
        </th>
        <th class=\\"\\" focusable=\\"false\\" scope=\\"col\\">
          <div class=\\"slds-truncate\\">Amount</div>
        </th>
        <th class=\\"\\" focusable=\\"false\\" scope=\\"col\\">
          <div class=\\"slds-truncate\\">Contact</div>
        </th>
      </tr>
    </thead>
    <tbody>
      <tr class=\\"\\">
        <td>
          <div class=\\"\\" title=\\"Cloudhub\\"><a href=\\"javascript:void(0);\\">Cloudhub</a></div>
        </td>
        <td>
          <div class=\\"\\" title=\\"Cloudhub\\">Cloudhub</div>
        </td>
        <td>
          <div class=\\"\\" title=\\"4/14/2015\\">4/14/2015</div>
        </td>
        <td>
          <div class=\\"\\" title=\\"Prospecting\\">Prospecting</div>
        </td>
        <td>
          <div class=\\"\\" title=\\"20%\\">20%</div>
        </td>
        <td>
          <div class=\\"\\" title=\\"$25k\\">$25k</div>
        </td>
        <td>
          <div class=\\"\\" title=\\"jrogers@cloudhub.com\\"><a href=\\"javascript:void(0);\\">jrogers@cloudhub.com</a></div>
        </td>
      </tr>
      <tr class=\\"\\">
        <td>
          <div class=\\"\\" title=\\"Cloudhub + Anypoint Connectors\\"><a href=\\"javascript:void(0);\\">Cloudhub + Anypoint Connectors</a></div>
        </td>
        <td>
          <div class=\\"\\" title=\\"Cloudhub\\">Cloudhub</div>
        </td>
        <td>
          <div class=\\"\\" title=\\"4/14/2015\\">4/14/2015</div>
        </td>
        <td>
          <div class=\\"\\" title=\\"Prospecting\\">Prospecting</div>
        </td>
        <td>
          <div class=\\"\\" title=\\"20%\\">20%</div>
        </td>
        <td>
          <div class=\\"\\" title=\\"$25k\\">$25k</div>
        </td>
        <td>
          <div class=\\"\\" title=\\"jrogers@cloudhub.com\\"><a href=\\"javascript:void(0);\\">jrogers@cloudhub.com</a></div>
        </td>
      </tr>
      <tr class=\\"\\">
        <td>
          <div class=\\"\\" title=\\"Cloudhub\\"><a href=\\"javascript:void(0);\\">Cloudhub</a></div>
        </td>
        <td>
          <div class=\\"\\" title=\\"Cloudhub\\">Cloudhub</div>
        </td>
        <td>
          <div class=\\"\\" title=\\"4/14/2015\\">4/14/2015</div>
        </td>
        <td>
          <div class=\\"\\" title=\\"Prospecting\\">Prospecting</div>
        </td>
        <td>
          <div class=\\"\\" title=\\"20%\\">20%</div>
        </td>
        <td>
          <div class=\\"\\" title=\\"$25k\\">$25k</div>
        </td>
        <td>
          <div class=\\"\\" title=\\"jrogers@cloudhub.com\\"><a href=\\"javascript:void(0);\\">jrogers@cloudhub.com</a></div>
        </td>
      </tr>
    </tbody>
  </table>
  <h3 class=\\"slds-text-heading--medium slds-m-vertical--medium\\">Column Bordered</h3>
  <table class=\\"slds-table slds-table--bordered slds-table--cell-buffer slds-table--col-bordered\\" id=\\"DataTableExample-columnBordered\\">
    <thead>
      <tr class=\\"slds-text-title--caps\\">
        <th class=\\"\\" focusable=\\"false\\" scope=\\"col\\">
          <div class=\\"slds-truncate\\">Opportunity Name</div>
        </th>
        <th class=\\"\\" focusable=\\"false\\" scope=\\"col\\">
          <div class=\\"slds-truncate\\">Account Name</div>
        </th>
        <th class=\\"\\" focusable=\\"false\\" scope=\\"col\\">
          <div class=\\"slds-truncate\\">Close Date</div>
        </th>
        <th class=\\"\\" focusable=\\"false\\" scope=\\"col\\">
          <div class=\\"slds-truncate\\">Stage</div>
        </th>
        <th class=\\"\\" focusable=\\"false\\" scope=\\"col\\">
          <div class=\\"slds-truncate\\">Confidence</div>
        </th>
        <th class=\\"\\" focusable=\\"false\\" scope=\\"col\\">
          <div class=\\"slds-truncate\\">Amount</div>
        </th>
        <th class=\\"\\" focusable=\\"false\\" scope=\\"col\\">
          <div class=\\"slds-truncate\\">Contact</div>
        </th>
      </tr>
    </thead>
    <tbody>
      <tr class=\\"\\">
        <td>
          <div class=\\"\\" title=\\"Cloudhub\\"><a href=\\"javascript:void(0);\\">Cloudhub</a></div>
        </td>
        <td>
          <div class=\\"\\" title=\\"Cloudhub\\">Cloudhub</div>
        </td>
        <td>
          <div class=\\"\\" title=\\"4/14/2015\\">4/14/2015</div>
        </td>
        <td>
          <div class=\\"\\" title=\\"Prospecting\\">Prospecting</div>
        </td>
        <td>
          <div class=\\"\\" title=\\"20%\\">20%</div>
        </td>
        <td>
          <div class=\\"\\" title=\\"$25k\\">$25k</div>
        </td>
        <td>
          <div class=\\"\\" title=\\"jrogers@cloudhub.com\\"><a href=\\"javascript:void(0);\\">jrogers@cloudhub.com</a></div>
        </td>
      </tr>
      <tr class=\\"\\">
        <td>
          <div class=\\"\\" title=\\"Cloudhub + Anypoint Connectors\\"><a href=\\"javascript:void(0);\\">Cloudhub + Anypoint Connectors</a></div>
        </td>
        <td>
          <div class=\\"\\" title=\\"Cloudhub\\">Cloudhub</div>
        </td>
        <td>
          <div class=\\"\\" title=\\"4/14/2015\\">4/14/2015</div>
        </td>
        <td>
          <div class=\\"\\" title=\\"Prospecting\\">Prospecting</div>
        </td>
        <td>
          <div class=\\"\\" title=\\"20%\\">20%</div>
        </td>
        <td>
          <div class=\\"\\" title=\\"$25k\\">$25k</div>
        </td>
        <td>
          <div class=\\"\\" title=\\"jrogers@cloudhub.com\\"><a href=\\"javascript:void(0);\\">jrogers@cloudhub.com</a></div>
        </td>
      </tr>
      <tr class=\\"\\">
        <td>
          <div class=\\"\\" title=\\"Cloudhub\\"><a href=\\"javascript:void(0);\\">Cloudhub</a></div>
        </td>
        <td>
          <div class=\\"\\" title=\\"Cloudhub\\">Cloudhub</div>
        </td>
        <td>
          <div class=\\"\\" title=\\"4/14/2015\\">4/14/2015</div>
        </td>
        <td>
          <div class=\\"\\" title=\\"Prospecting\\">Prospecting</div>
        </td>
        <td>
          <div class=\\"\\" title=\\"20%\\">20%</div>
        </td>
        <td>
          <div class=\\"\\" title=\\"$25k\\">$25k</div>
        </td>
        <td>
          <div class=\\"\\" title=\\"jrogers@cloudhub.com\\"><a href=\\"javascript:void(0);\\">jrogers@cloudhub.com</a></div>
        </td>
      </tr>
    </tbody>
  </table>
</div>"
`;
